x-pack/plugins/apm/public/components/app/mobile/transaction_overview/transaction_overview_tabs/use_mobile_statistics_fetcher.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { number } from 'joi';
+import { v4 as uuidv4 } from 'uuid';
+import { useApmServiceContext } from '../../../../../context/apm_service/use_apm_service_context';
+import { useFetcher } from '../../../../../hooks/use_fetcher';
+import { isTimeComparison } from '../../../../shared/time_comparison/get_comparison_options';
+
+const INITIAL_STATE_MAIN_STATISTICS = {
+  mainStatistics: [],
+  requestId: undefined,
+  totalItems: number,
+};
+
+const INITIAL_STATE_DETAILED_STATISTICS = {
+  currentPeriod: {},
+  previousPeriod: {},
+};
+
+interface Props {
+  field: string;
+  environment: string;
+  start: string;
+  end: string;
+  kueryWithMobileFilters: string;
+  comparisonEnabled: boolean;
+  offset?: string;
+}
+
+export function useMobileStatisticsFetcher({
+  field,
+  environment,
+  start,
+  end,
+  kueryWithMobileFilters,
+  comparisonEnabled,
+  offset,
+}: Props) {
+  const { serviceName } = useApmServiceContext();
+
+  const { data = INITIAL_STATE_MAIN_STATISTICS, status: mainStatisticsStatus } =
+    useFetcher(
+      (callApmApi) => {
+        if (start && end) {
+          return callApmApi(
+            'GET /internal/apm/mobile-services/{serviceName}/main_statistics',
+            {
+              params: {
+                path: { serviceName },
+                query: {
+                  environment,
+                  kuery: kueryWithMobileFilters,
+                  start,
+                  end,
+                  field,
+                },
+              },
+            }
+          ).then((response) => {
+            return {
+              // Everytime the main statistics is refetched, updates the requestId making the comparison API to be refetched.
+              requestId: uuidv4(),
+              mainStatistics: response.mainStatistics,
+              totalItems: response.mainStatistics.length,
+            };
+          });
+        }
+      },
+      [environment, start, end, kueryWithMobileFilters, serviceName, field]
+    );
+
+  const { mainStatistics, requestId, totalItems } = data;
+
+  const {
+    data: detailedStatistics = INITIAL_STATE_DETAILED_STATISTICS,
+    status: detailedStatisticsStatus,
+  } = useFetcher(
+    (callApmApi) => {
+      if (totalItems > 0 && start && end) {
```

Comment:
when `totalItems < 0` `detailedStatisticsStatus` will remain `FETCH_STATUS.LOADING` in some cases. Is this properly handled in the UI? if not we will end up in an infinite loading state (for references you can check [this PR](https://github.com/elastic/kibana/pull/150337))

Reply:
This was the hack I mentioned in my previous comment, now I just check if `totaltems` is not undefined, without this the detailed statistics call is done twice and the first one is going to fail, I checked how this was fixed on several places where we do the main and detailed statistics requests